app/services/idv/steps/welcome_step.rb

```diff
@@ -6,12 +6,26 @@ def call
       end
 
       def form_submit
+        skip_to_capture if params[:skip_upload] && FeatureManagement.document_capture_step_enabled?
+
         Idv::ConsentForm.new.submit(consent_form_params)
       end
 
       def consent_form_params
         params.permit(:ial2_consent_given)
       end
+
+      private
+
+      def skip_to_capture
+        mark_step_complete(:upload)
+        mark_step_complete(:send_link)
+        mark_step_complete(:link_sent)
```

Comment:
Do we still need to mark these steps complete if we don't need to do them?

Reply:
> Do we still need to mark these steps complete if we don't need to do them?

At least as I've understood the flow behavior to work, and observing that the doc auth flow is [implemented as including all possible steps](https://github.com/18F/identity-idp/blob/dc92fa4da95256711311f545a46080d1f049a148/app/services/idv/flows/doc_auth_flow.rb#L4-L20) (not dynamically generated based on features, device, etc), I think "Yes". Otherwise, those steps would be considered incomplete, and the user forced to complete it.

The one thing I wasn't quite as certain of here is how best to handle the fact that we have mobile and desktop flows, and for each, an old flow (`front_image`, `back_image`, `mobile_front_image`, `mobile_back_image`) and a new flow (`document_capture`, `mobile_document_capture`), plus assuming that this "Skip to Capture" only applies for mobile, and not desktop, and that related to this effort we want to consolidate new mobile and desktop flows (LG-3205).

Many factors to consider 😄 

- We can assume `front_image` and `back_image` steps are skipped in new flow ([source](https://github.com/18F/identity-idp/blob/2f0f31f9ca0ed941fe511cd511269f8fc21325e5/app/services/idv/steps/doc_auth_base_step.rb#L162-L166))
- I skip `mobile_document_capture` as a preemptive assumption we'll want to consolidate to `document_capture` as the primary new flow
- Skipping `mobile_front_image` and `mobile_back_image` here might not be necessary, based on the [logic linked in the first bullet point](https://github.com/18F/identity-idp/blob/2f0f31f9ca0ed941fe511cd511269f8fc21325e5/app/services/idv/steps/doc_auth_base_step.rb#L162-L166)

Reply:
Ok! The steps code is not my expertise so thanks for explaining

Reply:
> * Skipping `mobile_front_image` and `mobile_back_image` here might not be necessary, based on the [logic linked in the first bullet point](https://github.com/18F/identity-idp/blob/2f0f31f9ca0ed941fe511cd511269f8fc21325e5/app/services/idv/steps/doc_auth_base_step.rb#L162-L166)

Removed in rebased b745250.